controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -107,237 +107,70 @@ func ReadEnv() string {
 	return cEnv
 }
 
-//Reconcile fn
-func (r *ClowdEnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+//Acts as a guard for a reconciliation cycle, as well as initatizes a bunch of required objects
+func (r *ClowdEnvironmentReconciler) initReconciliation(ctx context.Context, req ctrl.Request) (logr.Logger, context.Context, crd.ClowdEnvironment, ctrl.Result, error) {
 	log := r.Log.WithValues("env", req.Name).WithValues("rid", utils.RandString(5))
 	ctx = context.WithValue(ctx, errors.ClowdKey("log"), &log)
 	ctx = context.WithValue(ctx, errors.ClowdKey("recorder"), &r.Recorder)
 	env := crd.ClowdEnvironment{}
 
+	silentFailError := errors.New(SILENTFAIL)
```

Comment:
Why pass this back at all? Why not just pass back nil? Then there is no need to do the conditional later in the `Reconcile` function

Reply:
Yeah that's a little complex. I'll explain why it is that way and then let me know if you see a better way to handle it.

The core problem here is that the Reconcile function takes 3 different actions depending on the result of attempting to get the env from the client:
- If there's no error move on with the reconcile
- If the error is that the environment was deleted return an empty result and nil, terminating the reconcile but not returning an error
- If the error is any other error then terminate the reconcile and return the error

So the case of the env having been deleted is a special case where we silently fail. That's how it worked in the original code:

```go
if getEnvErr := r.Client.Get(ctx, req.NamespacedName, &env); getEnvErr != nil {
		if k8serr.IsNotFound(getEnvErr) {
			// Must have been deleted
			return ctrl.Result{}, nil
		}
		log.Info("Namespace not found", "err", getEnvErr)
		return ctrl.Result{}, getEnvErr
	}
```

So when I factored this out into a method I had to provide some way to tell the Reoncile function "Hey this is the special case where you want to bail based on an error but where you don't want to return the error. I didn't see any other way to handle this, and looking at the code I still don't (except to forgo using the special SILENTFAIL error I created and just move the `k8serr.IsNotFound` check into the error handler in Reconcile.

So open to suggestions on this one for sure.